### ISSUE TITLE
False Positive InjectDispatcher

### ISSUE THREAD

**Issue by jonas-youniqx** [detekt/detekt#6200]
## Expected Behavior

Injected dispatchers get recognised as injected and not reported by detekt.

## Observed Behavior

> Dispatcher IO is used without dependency injection. [InjectDispatcher]

## Steps to Reproduce

We use an interface to inject all dispatchers via koin:

```kotlin
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.Dispatchers as KDispatchers

@Suppress("PropertyName", "VariableNaming")
interface IDispatchers {
    val Default: CoroutineDispatcher
    val Main: CoroutineDispatcher
    val Unconfined: CoroutineDispatcher
    val IO: CoroutineDispatcher
}

@Suppress("InjectDispatcher")
object Dispatchers : IDispatchers {
    override val Default = KDispatchers.Default
    override val Main = KDispatchers.Main
    override val Unconfined = KDispatchers.Unconfined
    override val IO = KDispatchers.IO
}
```

Example usage, I also tried to change it to constructor injection but the issue persists:
```kotlin
class Example : KoinComponent {
    private val dispatchers: IDispatchers by inject()
​
    suspend fun loadHtml(url: String): LoadHtmlResponse {
        return withContext(dispatchers.IO) { // <--  Dispatcher IO is used without dependency injection. [InjectDispatcher]
            ...
```
According to the code of the rule and my tries I assume that the issue is that we inject a whole bundle of dispatchers and access a property of this bundle directly.

## Context

Currently I disabled the rule. Would love to enable it again.

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): gradle-8.1.1 / androidGradlePlugin = "7.4.2"
* Gradle scan link (add `--scan` option when running the gradle task): -
* Operating System and version: macOS Ventura 13.4
* Link to your project (if it's a public repository): -

**Comment by cortinico** [comment]
> I also tried to change it to constructor injection but the issue persists:

The fact that the issue persists if you use constructor injection, is a bug in the rule that we should fix.

**Comment by atulgpt** [comment]
Hi @cortinico, @marschwar should we not issue a warning when dispatchers are coming with the delegated property?

**Comment by cortinico** [comment]
> should we not issue a warning when dispatchers are coming with the delegated property?

Delegated properties are fine imho as that's how DI frameworks might actually inject their dependencies

**Comment by atulgpt** [comment]
Hi, @jonas-youniqx I tried to run below TC and it is passing successfullly

```kotlin
@Test
fun `does not report when dispatcher is used via delegation`() {
    val code = """
        import kotlinx.coroutines.CoroutineDispatcher
        import kotlinx.coroutines.launch
        import kotlinx.coroutines.runBlocking
        import kotlinx.coroutines.Dispatchers
        import kotlin.reflect.KProperty

        interface IDispatchers {
            val io: CoroutineDispatcher
        }

        @Suppress("InjectDispatcher")
        object Dispatchers : IDispatchers {
            override val IO = Dispatchers.IO
        }

        class Inject {
            operator fun getValue(thisRef: Any?, property: KProperty<*>): Dispatchers {
                TODO("not implemented")
            }
        }

        fun useDispatchers() {
            val dispatchers by Inject()
            runBlocking {
                launch(dispatchers.io) { }
            }
        }
    """.trimIndent()
    assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
}
```

I am able to use the `dispatchers.io` in the `launch` via delegated method `Inject()`

**Comment by eygraber** [comment]
I'm seeing a false positive in some cases, but not others.
